apitools/base/py/exceptions.py

```diff
@@ -201,3 +201,43 @@ class NotYetImplementedError(GeneratedClientError):
 class StreamExhausted(Error):
 
     """Attempted to read more bytes from a stream than were available."""
+
+
+class UnrecognizedDataError(InvalidDataError):
```

Comment:
I'm leaning towards moving this out of apitools and just providing the interface necessary to build it (i.e. UnrecognizedFieldIter).

Ultimately, the question is how much of a compiler-type role apitools should seek to fill? I think it should avoid opinions about error formatting but provide all the data and tools necessary to produce user-friendly errors.